lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -3,49 +3,478 @@
-    uint32_t BytesWrittenPtr [[maybe_unused]]) {
-  return 0;
+Expect<uint32_t>
+WasiNNGetOuput::body(Runtime::Instance::MemoryInstance *MemInst,
+                     uint32_t Context, uint32_t Index [[maybe_unused]],
+                     uint32_t OutBuffer [[maybe_unused]],
+                     uint32_t OutBufferMaxSize [[maybe_unused]],
+                     uint32_t BytesWrittenPtr [[maybe_unused]]) {
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::ExecutionFailed);
+  }
+
+  if (Ctx.GraphContextBackends.size() <= Context) {
+    spdlog::error("[WASI-NN] get_output: Execution Context does not exist");
+    return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+  }
+
+  if (Ctx.GraphContextBackends[Context] == Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    uint8_t *OutBufferPtr;
+    uint32_t *BytesWritten;
+    OutBufferPtr = MemInst->getPointer<uint8_t *>(OutBuffer, 1);
+    BytesWritten = MemInst->getPointer<uint32_t *>(BytesWrittenPtr, 1);
+
+    IEStatusCode Status;
+    WASINN::OpenVINOSession *Session = Ctx.OpenVINOInfers[Context];
+    ie_network_t *Network = Session->Network;
+    ie_infer_request_t *InferRequest = Session->InferRequest;
+    ie_blob_t *OutputBlob = nullptr;
+    ie_blob_buffer_t BlobCBuffer;
+    char *OutputName = nullptr;
+    int BlobSize;
+
+    Status = ie_network_get_output_name(Network, Index, &OutputName);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error(
+          "[WASI-NN] Unable to find output name correctly with Index:{}",
+          Index);
+      ie_network_name_free(&OutputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    Status =
+        ie_network_set_output_precision(Network, OutputName, precision_e::FP32);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error(
+          "[WASI-NN] Unable to set output precision correctly with Index:{}",
+          Index);
+      ie_network_name_free(&OutputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+
+    Status = ie_infer_request_get_blob(InferRequest, OutputName, &OutputBlob);
+    if (Status != IEStatusCode::OK || OutputBlob == nullptr) {
+      spdlog::error("[WASI-NN] Unable to retrieve output tensor correctly",
+                    Index);
+      ie_network_name_free(&OutputName);
+      ie_blob_free(&OutputBlob);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    Status = ie_blob_size(OutputBlob, &BlobSize);
+    Status = ie_blob_get_cbuffer(OutputBlob, &BlobCBuffer);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to retrieve output tensor correctly",
+                    Index);
+      ie_network_name_free(&OutputName);
+      ie_blob_free(&OutputBlob);
+      return static_cast<uint32_t>(WASINN::ErrNo::MissingMemory);
+    }
+    float *BlobData =
+        const_cast<float *>(static_cast<float const *>(BlobCBuffer.cbuffer));
```

Comment:
The `ie_blob_buffer` is a `union`:
https://docs.openvino.ai/cn/2021.1/ie_c_api/structie__blob__buffer.html

Why don't you use the `BlobCBuffer.buffer` directly instead of using `const_cast`?